### DISCUSSION TITLE
Helping with documentation

### DISCUSSION THREAD

**Discussion by tido-** [jonaswinkler/paperless-ng#456] [General]
Hi,

It feels and looks much better now (I installed the 'dev' container) yesterday. 
The only thing I thought was strange is, to return from the Admin-Page, the logo doesn't take you home.

ARM | Rock Pi 4B - by the way, it flies on that SBC (single board computer).

I would like to improve the installation documentation, I walked every step and some are missing (docker).
Also I would like to add the information that if you intend to run it on a server, not on your PC to change localhost to 0.0.0.0:8000 so that it is reachable in your internal network.

Which branch shall I take to edit?

**Comment by jonaswinkler** [comment]
> It feels and looks much better now (I installed the 'dev' container) yesterday.

Just be aware that I sometimes push changes onto that branch that aren't quite ready / done yet :)

> The only thing I thought was strange is, to return from the Admin-Page, the logo doesn't take you home.

Which logo exactly? There's a "View Site" link in the top right to go back.

> ARM | Rock Pi 4B - by the way, it flies on that SBC (single board computer).

Glad to hear that!

> I would like to improve the installation documentation, I walked every step and some are missing (docker).
> Also I would like to add the information that if you intend to run it on a server, not on your PC to change localhost to 0.0.0.0:8000 so that it is reachable in your internal network.

Oh, when you run the docker image as per the instructions, it will bind to 0.0.0.0:8000; I can reach my instance just fine without any modifications. I think the documentation doesn't mention that.

> Which branch shall I take to edit?

Changes to the documentation are fine on the `master` branch.

**Comment by tido-** [comment]
Sure, but I wanted to see the latest development :)  

Logo - is maybe too much, left hand top corner "paperless-ng" on the admin page links to http://x.x.x.x:8000/admin/  I just thought to dashboard would be easier to return than click on the text:  Auf der Website anzeigen (on the right hand side)
Or seeing your text, the german is komisch ;) well thinking. Homepage  or  Startpage  would fit both languages.

But the .yml here points to localhost: https://github.com/jonaswinkler/paperless-ng/blob/dev/docker/compose/docker-compose.postgres.yml
Where do you change that to 0.0.0.0. ?   Oh gosh, it was too late when I tested it. 2 days ago and now thinking about it, the container didn't come up - so I guess this was the problem, not the localhost.

**Comment by jonaswinkler** [comment]
> Logo - is maybe too much, left hand top corner "paperless-ng" on the admin page links to http://x.x.x.x:8000/admin/ I just thought to dashboard would be easier to return than click on the text: Auf der Website anzeigen (on the right hand side)
> Or seeing your text, the german is komisch ;) well thinking. Homepage or Startpage would fit both languages.

The entire layout and text on the admin pages is provided by the development framework and I have little control over the things displayed there. I also don't care too much about the admin, it's supposed to be functional, which it is.

> But the .yml here points to localhost: https://github.com/jonaswinkler/paperless-ng/blob/dev/docker/compose/docker-compose.postgres.yml

if you refer to the part that says `healthcheck`, that is simply an instruction for docker and specifies how to check if paperless is (still) running. It does not specify where paperless is served at all.

> Where do you change that to 0.0.0.0. ? Oh gosh, it was too late when I tested it. 2 days ago and now thinking about it, the container didn't come up - so I guess this was the problem, not the localhost.

I'm not sure what problem you are referring to.

**Reply by darkmattercoder** [reply]
Regarding to the obviously wrong translation in the admin area, which @tido- referred to, I also was bothered by that, so maybe this would lead to a fix: https://github.com/django/django/pull/15319

**Comment by tido-** [comment]
Feedback on the ARM front - Raspberry Pi 2 - fresh install of raspios lite (for server), SDcard 32GB Samsung Evo+
The paperless-ng docker install was smooth, and faster than I expected.
```
$ docker --version   (from Docker website)
Docker version 20.10.2, build 2291f61
$ docker-compose --version  (from Debian Repo)
docker-compose version 1.21.0
```
```
Log file of a supermarket slip

26. Januar 2021 21:45 | Document 2020-02-01 Scan DENNER 2020-02-01 18.41.48 consumption finished | Information
  | Deleting directory /tmp/paperless/paperless-5pn0tmjk | Debug
  | Deleting file /tmp/paperless/paperless-upload-u0b3hsm7 | Debug
  | Saving record to database | Debug
  | Cannot classify documents: [Errno 2] No such file or directory: '/usr/src/paperless/src/../data/classification_model.pickle'. | Warnung
  | Execute:  optipng -silent -o5 /tmp/paperless/paperless-5pn0tmjk/convert.png -out  /tmp/paperless/paperless-5pn0tmjk/thumb_optipng.png | Debug
  | Execute:  convert -density 300 -scale 500x5000> -alpha remove -strip  -auto-orient /tmp/paperless/paperless-upload-u0b3hsm7[0]  /tmp/paperless/paperless-5pn0tmjk/convert.png | Debug
  | Generating thumbnail for Scan DENNER 2020-02-01 18.41.48.pdf... | Debug
  | Calling  OCRmyPDF with {'input_file':  '/tmp/paperless/paperless-upload-u0b3hsm7', 'output_file':  '/tmp/paperless/paperless-5pn0tmjk/archive.pdf', 'use_threads': True,  'jobs': 3, 'language': 'deu', 'output_type': 'pdfa', 'progress_bar':  False, 'clean': True, 'redo_ocr': True} | Debug
  | No text was found in the document and skip is specified. Upgrading OCR mode to redo. | Debug
  | Parsing Scan DENNER 2020-02-01 18.41.48.pdf... | Debug
  | Parser: RasterisedDocumentParser | Debug
  | Detected mime type: application/pdf | Debug
26. Januar 2021 21:40 | Consuming Scan DENNER 2020-02-01 18.41.48.pdf | Information
```
While running OCR it still has free memory
```
$ free -m
              total        used        free      shared  buff/cache   available
Mem:            924         436         276          15         212         421
Swap:            99          28          71
```
The web interface works fast. Doing Tagging, document type, correspondent happens almost immediately I could easily life with it. The A4 Paper and the supermarket slip took both OCR around 6 minutes. This with only 3-5 Watt power consumption.

I hope this helps one or the other.

**Comment by jonaswinkler** [comment]
Well, OCR is expensive and the Pi 2 is not particularly fast, so that's to be expected.

**Reply by tido-** [reply]
Sure, but there was such a big fuss if NG will work on RPi, so it was time for a test. And as I only have a RPi 2, I gave it a try and tested the manual as well :)

Something else, https://paperless-ng.readthedocs.io/en/latest/setup.html#install-paperless-from-docker-hub  on step 4. you suggest to adjust the "consume" folder, but if people do not adjust the location of the DB storage folder and some day update (replace) the docker container, wont they lose everything?

**Reply by jonaswinkler** [reply]
> Sure, but there was such a big fuss if NG will work on RPi, so it was time for a test. And as I only have a RPi 2, I gave it a try and tested the manual as well :)

Might be worth mentioning in the thread over at the paperless project, but i'm not sure how many people read that anyway.

> Something else, https://paperless-ng.readthedocs.io/en/latest/setup.html#install-paperless-from-docker-hub on step 4. you suggest to adjust the "consume" folder, but if people do not adjust the location of the DB storage folder and some day update (replace) the docker container, wont they lose everything?

No, since containers and volumes are separate things. You can remove the paperless container, even the image, and the volume will still be there. When you download the image again and start a new container, it will still use the existing images, given that the project name in the .env file is the same.

The current compose files will create a volume managed by docker for both data and media, and mount that into a container when started.

People will only loose their files if they decide to stop and remove the paperless container (safe) and then also decide to remove the volume explicitly (not safe), or delete their /var/lib/docker/volumes folder.
